docs/arch.md

```diff
@@ -0,0 +1,286 @@
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2020-10-03T04:14:05Z",
+    "status": "False",
+    "type": "Disabled"
+  }
+]
+```
+The status is being updated by `pkg/controller/status/status.go`. Status has a background task, which is periodically updating 
+the Operator status from its internal list of Sources. Any component which wants to participate on Operator's status adds a
+SimpleReporter, which is returning its actual Status. The Simple reporter is defined in controllerstatus.
+
+Code: In `operator.go` components are adding their reporters to Status Sources:
+```
+statusReporter.AddSources(uploader)
+```
+This periodic status updater calls updateStatus which sets the Operator status after calling merge to all the provided Sources.
+The uploader updateStatus determines if it is Safe to upload, if Cluster Operator status and Pod last Exit Code are both healthy.
+It relies on fact that updateStatus is called on Start of status cycle.
+
+
+
+## How is Insights Operator using various Api Clients
+Internally Insights operator is talking to Kubernetes Api server over Http Rest queries. Each query is authenticated by a Bearer token, 
+To simulate see an actual Rest query being used, you can try:
+```
+$ oc get pods -A -v=9
+I1006 12:26:33.972634   66541 loader.go:375] Config loaded from file:  /home/mkunc/.kube/config
+I1006 12:26:33.977546   66541 round_trippers.go:423] curl -k -v -XGET  -H "Accept: application/json;as=Table;v=v1;g=meta.k8s.io,application/json;as=Table;v=v1beta1;g=meta.k8s.io,application/json" -H "User-Agent: oc/4.5.0 (linux/amd64) kubernetes/9933eb9" -H "Authorization: Bearer Xy9HoVzNdsRifGr3oCIl7pfxwkeqE2u058avw6o969w" 'https://api.sharedocp4upi43.lab.upshift.rdu2.redhat.com:6443/api/v1/pods?limit=500'
+I1006 12:26:36.075230   66541 round_trippers.go:443] GET https://api.sharedocp4upi43.lab.upshift.rdu2.redhat.com:6443/api/v1/pods?limit=500 200 OK in 2097 milliseconds
+I1006 12:26:36.075284   66541 round_trippers.go:449] Response Headers:
+I1006 12:26:36.075300   66541 round_trippers.go:452]     Audit-Id: 53ad17b9-c3fe-4166-9693-2bacf60f7dcc
+I1006 12:26:36.075313   66541 round_trippers.go:452]     Cache-Control: no-cache, private
+I1006 12:26:36.075326   66541 round_trippers.go:452]     Content-Type: application/json
+I1006 12:26:36.075347   66541 round_trippers.go:452]     Vary: Accept-Encoding
+I1006 12:26:36.075370   66541 round_trippers.go:452]     Date: Tue, 06 Oct 2020 10:26:36 GMT
+I1006 12:26:36.467245   66541 request.go:1068] Response Body: {"kind":"Table","apiVersion":"meta.k8s.io/v1","metadata":{"selfLink":"/api/v1/pods"
+... CUT HERE
+```
+
+But adding Bearer token and creating Rest query is all handled automatically for us by using Clients, which are generated, type safe golang libraries, 
+like `[github.com/openshift/client-go](github.com/openshift/client-go)` or `[github.com/kubernetes/client-go](github.com/kubernetes/client-go)`.
```

Comment:
these links aren't highlighted(https://github.com/openshift/insights-operator/blob/844932597dce3973f0f78c8345ad150e6d74bf91/docs/arch.md#how-is-insights-operator-using-various-api-clients), probably there shouldn't be `